common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java

```diff
@@ -0,0 +1,224 @@
+            return c.doFinal(out);
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (final NoSuchPaddingException e) {
+            errCode = NO_SUCH_PADDING;
+            exception = e;
+        } catch (final IllegalBlockSizeException e) {
+            errCode = INVALID_BLOCK_SIZE;
+            exception = e;
+        } catch (final BadPaddingException e) {
+            errCode = BAD_PADDING;
+            exception = e;
+        } catch (final InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        } catch (final InvalidAlgorithmParameterException e) {
+            errCode = INVALID_ALG_PARAMETER;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage(), exception);
+    }
+
+    @Override
+    public byte[] sign(@NonNull final byte[] text) throws ClientException {
+        final String errCode;
+        final Exception exception;
+        try {
+            final SecretKeySpec key = new SecretKeySpec(this.key, suite.cipher().name());
+            Mac mac = Mac.getInstance(suite.macName());
+            mac.init(key);
+            return mac.doFinal(text);
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (final InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage());
+    }
+
+    @Override
+    public boolean verify(@NonNull final byte[] text,
+                          @NonNull final byte[] signature) throws ClientException {
+        return Arrays.equals(signature, sign(text));
+    }
+
+    @Override
+    public byte[] getThumprint() throws ClientException {
+        final SecretKey keySpec = new SecretKeySpec(key, suite.cipher().name());
+        final Cipher cipher;
+        final String errCode;
+        final Exception exception;
+        try {
+            cipher = Cipher.getInstance(keySpec.getAlgorithm());
+            final MessageDigest digest = MessageDigest.getInstance("SHA256");
+            return digest.digest(cipher.doFinal((keySpec.getAlgorithm() + cipher.getBlockSize() + cipher.getParameters()).getBytes(UTF8)));
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (final NoSuchPaddingException e) {
+            errCode = NO_SUCH_PADDING;
+            exception = e;
+        } catch (final BadPaddingException e) {
+            errCode = BAD_PADDING;
+            exception = e;
+        } catch (final IllegalBlockSizeException e) {
+            errCode = INVALID_BLOCK_SIZE;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage(), exception);
+    }
+
+    @Override
```

Comment:
```suggestion
    @Override
    @Nullable
```

Idk maybe this annot is pointless on the subclass 🤷‍♂️

Reply:
(This might need an `import` too, visually scanning this file)